### ISSUE TITLE
Exceptions when stopping server

### ISSUE THREAD

**Issue by Akiranya** [WiIIiam278/HuskSync#95]
Sometimes when I stop the server I got this error

tho I haven't see any noticeable issues in-game regarding to my inventory/advancements/etc. yet

# Exceptions

```
21:23:18 INFO]: [com.zaxxer.hikari.HikariDataSource] HikariPool-2 - Shutdown initiated...                                                                                             
[21:23:18 INFO]: [com.zaxxer.hikari.HikariDataSource] HikariPool-2 - Shutdown completed.                                                                                               
[21:23:18 INFO]: [HuskSync] Disabling HuskSync v2.2.3         
[21:23:18 ERROR]: [HuskSync] Failed to get user data from online player Nailm (org.bukkit.plugin.IllegalPluginAccessException: Plugin attempted to register task while disabled)
[21:23:18 WARN]: java.util.concurrent.CompletionException: org.bukkit.plugin.IllegalPluginAccessException: Plugin attempted to register task while disabled
[21:23:18 WARN]:        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
[21:23:18 WARN]:        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
[21:23:18 WARN]:        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770)
[21:23:18 WARN]:        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)                                                              
[21:23:18 WARN]:        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)                                                                                   
[21:23:18 WARN]:        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
[21:23:18 WARN]:        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
[21:23:18 WARN]:        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
[21:23:18 WARN]:        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
[21:23:18 WARN]: Caused by: org.bukkit.plugin.IllegalPluginAccessException: Plugin attempted to register task while disabled
[21:23:18 WARN]:        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftScheduler.validate(CraftScheduler.java:552)
[21:23:18 WARN]:        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftScheduler.runTaskTimer(CraftScheduler.java:227)                                                              
[21:23:18 WARN]:        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftScheduler.runTaskTimer(CraftScheduler.java:212)
[21:23:18 WARN]:        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftScheduler.runTaskLater(CraftScheduler.java:176)
[21:23:18 WARN]:        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftScheduler.runTask(CraftScheduler.java:145)                                                                   
[21:23:18 WARN]:        at HuskSync-2.2.3.jar//net.william278.husksync.player.BukkitPlayer.getPersistentDataContainer(BukkitPlayer.java:442)
[21:23:18 WARN]:        at HuskSync-2.2.3.jar//net.william278.husksync.player.OnlineUser$2.<init>(OnlineUser.java:352)
[21:23:18 WARN]:        at HuskSync-2.2.3.jar//net.william278.husksync.player.OnlineUser.lambda$getUserData$3(OnlineUser.java:324)
[21:23:18 WARN]:        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
[21:23:18 WARN]:        ... 6 more                                                         
[21:23:18 INFO]: [net.william278.husksync.libraries.hikari.HikariDataSource] HuskSyncHikariPool - Shutdown initiated...
[21:23:18 INFO]: [net.william278.husksync.libraries.hikari.HikariDataSource] HuskSyncHikariPool - Shutdown completed.
[21:23:18 INFO]: [HuskSync] Successfully disabled HuskSync v2.2.3                                                                                                                      
[21:23:18 INFO]: [ExcellentEnchants] Disabling ExcellentEnchants v3.3.0.2
```

# Full server logs

https://pastes.dev/6zDOKDd8Uy

# Config

```
# ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
# ┃        HuskSync Config       ┃
# ┃    Developed by William278   ┃
# ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
# ┣╸ Information: https://william278.net/project/husksync
# ┗╸ Documentation: https://william278.net/docs/husksync
language: zh-cn
check_for_updates: false
cluster_id: ''
debug_logging: false
database:
  credentials:
    # Database connection settings
    host:
    port:
    database:
    username:
    password:
    parameters: '?autoReconnect=true&useSSL=false'
  connection_pool:
    # MySQL connection pool properties
    maximum_pool_size: 2
    minimum_idle: 2
    maximum_lifetime: 1800000
    keepalive_time: 0
    connection_timeout: 5000
  table_names:
    users: husksync_users
    user_data: husksync_user_data
redis:
  credentials:
    # Redis connection settings
    host:
    port:
    password:
  use_ssl: false
synchronization:
  # Synchronization settings
  max_user_data_snapshots: 5
  save_on_world_save: true
  save_on_death: true
  compress_data: true
  notification_display_slot: CHAT
  save_dead_player_inventories: true
  network_latency_milliseconds: 500
  features:
    experience: true
    max_health: true
    inventories: true
    locked_maps: false
    potion_effects: true
    game_mode: true
    location: false
    advancements: true
    ender_chests: true
    hunger: true
    persistent_data_container: true
    statistics: true
    health: true
  event_priorities:
    quit_listener: LOWEST
    join_listener: LOWEST
    death_listener: NORMAL
```

**Comment by Akiranya** [comment]
The error above only exists when I enabled PDC sync.
I also got this error when joining the server (if PDC is enabled):

```
[00:59:36 INFO]: UUID of player Nailm is 174eb4a8-20d1-4838-ab07-877cd1d68c14                                                              
[00:59:37 INFO]: Nailm[/113.247.216.25:44064] logged in with entity id 572 at ([city]103.4020187604535, 93.0, -522.3496485594557)     
[00:59:37 WARN]: [HuskSync] Could not write Nailm's persistent data map, skipping!                                                                                     
> java.util.concurrent.CompletionException: java.lang.ClassCastException: Cannot cast java.util.ArrayList to [Lorg.bukkit.persistence.PersistentDataContainer;
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)                                    
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)                                                              
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1807)              
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796)                                                                 
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)                                                                                   
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.helpAsyncBlocker(ForkJoinPool.java:1253)                                                              
        at java.base/java.util.concurrent.ForkJoinPool.helpAsyncBlocker(ForkJoinPool.java:2237)                                                                        
        at java.base/java.util.concurrent.CompletableFuture.waitingGet(CompletableFuture.java:1887)
        at java.base/java.util.concurrent.CompletableFuture.join(CompletableFuture.java:2117)   
        at HuskSync-2.2.3.jar//net.william278.husksync.player.OnlineUser.lambda$setData$2(OnlineUser.java:303)                                                         
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)                                                               
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)                                                              
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)                                                                                   
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)                                                                  
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)                                                                                    
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)                                                                               
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)                                                                      
Caused by: java.lang.ClassCastException: Cannot cast java.util.ArrayList to [Lorg.bukkit.persistence.PersistentDataContainer;                                          
        at java.base/java.lang.Class.cast(Class.java:3889)                         
        at HuskSync-2.2.3.jar//net.william278.husksync.data.PersistentDataContainerData.getTagValue(PersistentDataContainerData.java:32)                               
        at HuskSync-2.2.3.jar//net.william278.husksync.data.BukkitPersistentTypeMapping.setContainerValue(BukkitPersistentTypeMapping.java:40)
        at HuskSync-2.2.3.jar//net.william278.husksync.player.BukkitPlayer.lambda$setPersistentDataContainer$44(BukkitPlayer.java:477)                                 
        at java.base/java.util.Optional.ifPresentOrElse(Optional.java:196)                                                                                             
        at HuskSync-2.2.3.jar//net.william278.husksync.player.BukkitPlayer.lambda$setPersistentDataContainer$46(BukkitPlayer.java:477)                                 
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)                  
        at HuskSync-2.2.3.jar//net.william278.husksync.player.BukkitPlayer.lambda$setPersistentDataContainer$47(BukkitPlayer.java:472)                                 
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)                                                                  
        ... 14 more                      
[00:59:38 INFO]: [dev.lone.shaded.cosmeticscore.com.zaxxer.hikari.HikariDataSource] HikariPool-1 - Starting...                                                         
[00:59:38 INFO]: [dev.lone.shaded.cosmeticscore.com.zaxxer.hikari.pool.HikariPool] HikariPool-1 - Added connection com.mysql.cj.jdbc.ConnectionImpl@36a4c009
[00:59:38 INFO]: [dev.lone.shaded.cosmeticscore.com.zaxxer.hikari.HikariDataSource] HikariPool-1 - Start completed.                                                    
> awe                                    
[01:00:17 INFO]: 没有这个指令, 具体请查阅上古 wiki                                 
> fawe reload                            
[01:00:19 INFO]: WEPIF: Vault detected! Using Vault for permissions                
[01:00:19 INFO]: [com.fastasyncworldedit.core.Fawe] LZ4 Compression Binding loaded successfully                                                                        
[01:00:19 INFO]: [com.fastasyncworldedit.core.Fawe] ZSTD Compression Binding loaded successfully                                                                       
[01:00:19 INFO]: Configuration reloaded!
```

**Comment by WiIIiam278** [comment]
Fixed via ce3350c6fa508387da84fade590237e5b4dfae1a
